packages/manager/apps/web/client/app/hosting/general-informations/GENERAL_INFORMATIONS.controller.js

```diff
@@ -188,6 +190,34 @@ export default class HostingGeneralInformationsCtrl {
     return !this.hasSSLCertificate();
   }
 
+  canOrderCdn() {
+    const {
+      flushCdnState,
+      hosting: { hasCdn, offer },
+    } = this.$scope;
+
+    const isFlushCdnStateOk = flushCdnState === 'ok';
+    const isOfferStart10M = offer === 'START_10_M';
+
+    const canOrderCdn =
+      (isFlushCdnStateOk && !hasCdn && !isOfferStart10M) ||
+      (hasCdn && isFlushCdnStateOk) ||
+      hasCdn;
+
+    if (!canOrderCdn) {
+      return false;
+    }
+
+    const {
+      hosting: {
+        serviceInfos: { contactAdmin, contactBilling },
+      },
+    } = this.$scope;
+    const { nichandle } = this.user;
+
+    return [contactAdmin, contactBilling].includes(nichandle);
```

Comment:
May this logic be in separated method? This way you call this new method and combine its result with the previous return.
And then finish this method by `return canOrderCdn || this.hasRightsToOrderCdn()`

Reply:
:+1: but with a `&&` operator

Reply:
Yes I meant with `&&` ^^